### DISCUSSION TITLE
Simple example of Custom Dialog without MVVM

### DISCUSSION THREAD

**Discussion by Montago** [MahApps/MahApps.Metro#4329] [Q&A]
I've been sitting here for the past 2 hours trying to figure out how to show the in-app modal dialog.

The MahApps.Metro.Demo has some VERY convoluted and very hard to transfer examples of how to construct dialogs (why are they so?) - the MahApps.Metro.Demo codebase contains a ton of ExtensionMethods which only makes it even harder to figure out how they work.,......

The Documentation is almost non-existing : https://mahapps.com/docs/dialogs/login-dialogs
where only the MVVM contains SOME descriptions. 

And when trying to use the DialogCoordinator you also end up in a very consoluted situation where the Singleton needs a reference to the DataContext or viceverse - Waaaaaat (o_O)

Preferably it would be nice to do something like this: 

DialogService.ShowDialog( "titel", "text", Buttons.OkCancel)
DialogService.ShowDialog( new CustomUserControlInstance(), Buttons.OkCancel)

Why is this so hard ? 

Simple Question: 
**How do i present a Dialog that contains my own UserControl - that has an OK and Cancel button ?**

**Comment by punker76** [comment]
The documentation has now a section about this https://mahapps.github.io/mahapps.com/docs/dialogs/custom-dialogs
